### PR TITLE
Fix error when cls_pooling="mean" or cls_pooling="max" for TransformerDocumentEmbeddings

### PR DESCRIPTION
**Pull request by fkdosilovic**
* Refactor `"cls"` pooling option into a function.
* Return `result` from `document_mean_pooling` and `document_max_pooling` functions.
* Add unit tests for testing each cls pooling option.
* Refactor `test_default_embeddings_stay_the_same_after_saving_and_loading` and `test_embeddings_stay_the_same_after_saving_and_loading` in `BaseEmbeddingsTest` to use `torch.allclose`

Fixes https://github.com/flairNLP/flair/issues/3552.

### CODE REVIEW

File: tests/embedding_test_utils.py [hunk 1]
```diff
@@ -176,9 +184,17 @@ def test_default_embeddings_stay_the_same_after_saving_and_loading(self):
...
+                    token_old.get_embedding(names_old),
+                    token_new.get_embedding(names_new),
+                    atol=1e-6,
+                )
         if self.is_document_embedding:
-            assert (sentence_old.get_embedding(names_old) == sentence_new.get_embedding(names_new)).all()
+            assert torch.allclose(
+                sentence_old.get_embedding(names_old),
+                sentence_new.get_embedding(names_new),
+                atol=1e-6,
```
Comment (helpmefindaname) [inline tests/embedding_test_utils.py:196]: Can you please revert those changes?

The bug that transformer embeddings weren't the same after reloading are fixed on master now.

**Review by helpmefindaname — CHANGES_REQUESTED** [review]

File: tests/embedding_test_utils.py (hunk 1 above)
Comment (fkdosilovic) [reply tests/embedding_test_utils.py:196]: > Can you please revert those changes?

Just to clarify, do you want me to revert all the changes were I introduce `torch.allclose` or just the ones for `if self.is_document_embedding` branching?

Comment (helpmefindaname) [reply tests/embedding_test_utils.py:196]: yes exactly, the test is intended to ensure that embeddings work the same even when reloading them. Hence, it is expected to have the exact same values, as the same computer architecture is used

Comment (fkdosilovic) [reply tests/embedding_test_utils.py]: Done.

**Comment by helpmefindaname** [comment]
Thank you for the bugfix @fkdosilovic
I will merge as soon as the tests complete

**Review by helpmefindaname — APPROVED** [review]

**Pull request merged**